### ISSUE TITLE
Request to be able to type in Arabic in text fields

### ISSUE THREAD

**Issue by lbestock** [arch-kiosk/arch-kiosk-office#2651]
Coming out of ARCE 2024, Yaser says it is important to a) have as little typing in English as possible, more drop-down menus, and b) be able to type in Arabic for the description and particularly the journaling fields. He doesn't need the labels to be in Arabic, his concern is that his excavators be able to write fluently and easily their observations.

We wondered if this was easy.

I think it is:

![image](https://github.com/arch-kiosk/arch-kiosk-office/assets/49291909/baaca4c5-25b0-46ef-ad46-d0f5025bb43b)

(My Arabic is poor but I think this is proof of concept that one can type in Arabic on the iPad in FM already. It is just a matter of enabling the keyboard.)

Oh also, c) servers in Egypt, not at Brown. Bigger issue.

**Comment by lbestock** [comment]
However, it seems to do something weird to the images when I type in Arabic. Those dayplans had their images visible before, when I copied and pasted something in Arabic. Then I actually typed a sentence in Arabic (how many people still remember that I was called Suad, I wonder?) and they went away. So, still things to investigate about how Arabic and FM interact, but the basic ability is there.

**Comment by lbestock** [comment]
But proper support for RTL languages (right to left - the acronym problem is not limited to universities) including right collation seems to be something FM does not support and has stated explicitly that they won't support. So next question at this stage with Filemaker would be for Yaser and if this is enough for him at this stage.

Makes me more generally curious. Ancient languages are sometimes a directional mess - Egyptian can be written in a variety of ways. How do people transcribing Egyptian digitally deal with directionality?

**Comment by lbestock** [comment]
My brain is buzzing on this after talking with Reis Ashraf in Abydos. And then realizing that the people WhatsAppling me are often typing in Arabic in google.translate and then pasting the translation as the text. By the same means, I can text in Arabic. When I re-translate it is quite good. For projects that have internet at the house, translating narratives both ways as part of afternoon lab time would be five minutes of work and extremely beneficial - the Guftis, then, could read the notes of the supervisors, and could make notes of their own that the supervisors can read. It will help the archaeological communication between them during the work, and mean that the record then includes the direct observations of one of the most important people in the square.

If RTL failure in FM makes it too confusing to type (I'll talk to Yaser about this today with this example) then we just need a second app so the Gufti can type more easily, then copy and paste into FM. It would still display funny, as here, but I think not a big deal. 

![image](https://github.com/user-attachments/assets/1f2fe2d8-979e-423d-bda3-8ab61371b215)

**Comment by urapadmin** [comment]
Filemaker Pro does not allow RTL languages. Except if you buy the international version, distributed by a company called Winsoft. Current price per license: 634€. That is unfortunately only the FileMaker side. For Kiosk RTL is possible as it is Browser based but a whole new can of worms for the UI. I don't think I can handle that complexity as a one man band. 

But here you are even mixing two languages in one field and that would need a text processor in the little narrative field. That's definitely not going to happen. I can envision (purely conceptually) to have texts in several languages and then one can switch between them.

**Comment by urapadmin** [comment]
even in this example I would strongly advice you to insert the Arabic version into a second narrative.

**Comment by urapadmin** [comment]
the WinSoft Middle Eastern version might exist in an educational version: https://winsoft-international.com/forms/filemaker-education-nonprofit/ 

I read somewhere that you can only have a FM version that is either Arabic OR English but it won't be able to do both. I am not sure if that is accurate for current versions.

**Comment by urapadmin** [comment]
if an Arabic text is in its own field it would be possible, of course, to have the whole text being right aligned. The cursor still runs from left to right while typing, though. For just displaying things it should look more proper. For typing Arabic it is not making anything better.

**Comment by urapadmin** [comment]
all major translation services do have an API. So if there is an internet connection (bandwidth should not matter that much in this case) it would be possible to pull in translations automatically. Just want to note that an external app might not be necessary if one finds a good way to integrate that service in Kiosk or even the recording app itself.

**Comment by lbestock** [comment]
Yaser tonight showed me an app that translates colloquial Arabic quite well. What I'd been doing is formal Arabic, which is less than ideal. The naturalness that he wants to capture with observations in the trench prohibits formal almost as much as English. We couldn't finish the conversation, but we started it. Going Arabic to English with colloquial looks like it will be no problem, but English to colloquial we were not having nearly as much success with. It also matters much less in that direction since people can read formal Arabic more easily than they can compose texts in it. He'll send me information about the app. Pulling in translations automatically sounds amazing.

**Comment by lbestock** [comment]
The problem with having the translation in a different field is that then you lose the relationship between the original and the translation, which makes it harder to check and destroys the relationship with time, too. Most probably several different new narratives would be made before the translation, if it's being done in lab time.

**Comment by urapadmin** [comment]
no, that depends entirely on the way the fields are presented. I am rather thinking of ONE field on the UI, that can switch between languages. If you don't do it that way you create a blob of unicode characters that can't be parsed anymore by a computer. Deducing a language shift inside a text is problematic enough even if you don't take the fact into account that sometimes a language embeds a foreign language. You know, like "handy" and "flashi" etc. Not to speak about quotes, proper nouns etc. 
Ergo: There must be a clear distinction between languages. How one presents the separate fields on the UI is a different question.  

But again: We are leading an entirely theoretical debate here. This is not going to happen anytime this year.

**Comment by urapadmin** [comment]
see #3085
